### ISSUE TITLE
[Bug]: Regression: #35930 broke CJS via ESM loading when exporting null/undefined with ELECTRON_RUN_AS_NODE

### ISSUE THREAD

**Issue by Prinzhorn** [electron/electron#36970]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

22.0.3

### What operating system are you using?

Ubuntu

### Operating System Version

Ubuntu 22.10

### What arch are you using?

x64

### Last Known Working Electron version

20.3.1 / 21.1.0

### Expected Behavior

1. `npm i electron@20.3.1 ava`
2. `ELECTRON_RUN_AS_NODE=true ./node_modules/.bin/electron ./node_modules/.bin/ava`
3. Works (can't find any tests obviously)

I was successfully running my esm tests via `ELECTRON_RUN_AS_NODE` until now

### Actual Behavior

1. `npm i electron@20.3.2 ava`
2. `ELECTRON_RUN_AS_NODE=true ./node_modules/.bin/electron ./node_modules/.bin/ava`
3. Crashes

```
node:internal/modules/esm/translators:178
      if (!ObjectPrototypeHasOwnProperty(exports, exportName) ||
           ^

TypeError: Cannot convert undefined or null to object
    at hasOwnProperty (<anonymous>)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:178:12)
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:533:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12)
```

### Additional Information

I was able to track this down and repro without ava as well (https://github.com/avajs/ava/discussions/3155). It appears you can no longer export `null` because it crashes the esm loader patch.

index.mjs

```js
import m from './module.cjs';
```

module.cjs

```js
module.exports = null;
```

`ELECTRON_RUN_AS_NODE=true ./node_modules/.bin/electron index.mjs`

It broke between `20.3.1` and `20.3.2` because of #35930 . If you're curious, in ava this comes from this dep https://github.com/jamiebuilds/ci-parallel-vars/blob/994940e4882bdddf046a5c076b49d700449e4b38/index.js#L41-L56

> Weirdly, this is like the 4th time I've tried this. 

@MarshallOfSound fifth time is the charm

**Comment by Prinzhorn** [comment]
I think this is the culprit, because it assumes you can only export objects when in reality you can export anything.

```
Object.keys(module.exports)
```

**Comment by Abhijrathod** [comment]
(https://github.com/electron/electron/pull/35930) and is caused by a change in the way ESM are handled in electron. The issue can be resolved by downgrading the electron version to 20.3.1 or older, or by not exporting null from ESM.

**Comment by Prinzhorn** [comment]
Thanks for your reply.

1. I already quoted #35930 and mentioned the author of the PR
2. It's not a module I control (as mentioned it comes from `ci-parallel-vars` which `ava` uses), hence I cannot not export null (which is perfectly valid code)
3. Yes, I downgraded to `21.1.0` which is the most recent version without the patch
